### PR TITLE
mark env(s) as bad if node(s) cannot be accessible.

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/runners/lisa_runner.py [hunk 1]
```diff
@@ -352,6 +352,16 @@ def _run_test_task(
         # on next test cases, close the connection after each test run. It will
         # be connected on the next command automatically.
         environment.nodes.close()
+        # Try to connect node(s), if cannot access node(s) of this environment,
+        # set the current environment as Bad. So that this environment won't be reused.
+        if not environment.nodes.is_connectable():
```
Comment (squirrelsc) [inline lisa/runners/lisa_runner.py]: call `initialize` should be enough, it doesn't need to wrap a magic command.

File: lisa/sut_orchestrator/azure/common.py [hunk 2]
```diff
@@ -590,6 +590,19 @@ def check_or_create_resource_group(
                     resource_group_name, {"location": location}
                 )
 
+            timeout = 30
+            timeout_timer = create_timer()
+            while timeout_timer.elapsed(False) < timeout:
+                az_shared_rg_exists = rm_client.resource_groups.check_existence(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/common.py]: Is it fixed by wait longer? The create_or_update is a synchronized API as before.

File: microsoft/testsuites/network/common.py [hunk 3]
```diff
@@ -205,7 +205,7 @@ def cleanup_iperf3(environment: Environment) -> None:
 
 def remove_extra_nics(environment: Environment) -> None:
     for node in environment.nodes.list():
-        node = cast(RemoteNode, environment.nodes[0])
+        node = cast(RemoteNode, node)
```
Comment (squirrelsc) [inline microsoft/testsuites/network/common.py]: It's better to move the easy fixes in a separated PR.

**Comment by LiliDeng** [comment]
force rebase by accident.

**Pull request closed without merging**